### DISCUSSION TITLE
af::range fails with ERR_ARG

### DISCUSSION THREAD

**Discussion by jramapuram** [arrayfire/arrayfire-rust#25] [Q&A] [answered]
Code:

``` rust
  let dims = Dim4::new(&[128, 1, 1, 1]);
  af::print(&af::range(dims, 128, af::Aftype::F32).unwrap());
```

Result:

``` bash
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: ERR_ARG', src/libcore/result.rs:731
```

**Comment by jramapuram** [comment]
@9prady9 : Any thoughts? I don't see anything wrong with range() in data/mod.rs

**Comment by 9prady9** [comment, marked as answer]
@jramapuram Check the Rust doc for this function [here](http://arrayfire.github.io/arrayfire-rust/arrayfire/fn.range.html).
C++ documentation is [here](http://www.arrayfire.com/docs/group__data__func__range.htm)

The problem is with the second argument which is a dimension specifier - takes values 0, 1, 2, 3

I just created the skeleton documentation today and will keep updating the rust wrapper documentation once we figure how to use content (as much as possible) from C++ docs in rust docs.

---

Assuming you wanted to create an array of size 128x128 with sequenced values along 1st dimension.
it should be the following

```
af::print(&af::range(dims, 1, af::Aftype::F32).unwrap());
```

that would give the following output

```
0 1 2 .... 127
0 1 2 .... 127
0 1 2 .... 127
.
.
.
0 1 2 .... 127
```

**Comment by jramapuram** [comment]
Ah! Thanks, I thought it was the initial value to start the range from. 
Nice job with the docs too, will be helpful!
